modules/gapi/samples/oak_basic_infer.cpp

```diff
@@ -0,0 +1,132 @@
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+
+#include <opencv2/imgproc.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/gapi.hpp>
+#include <opencv2/gapi/core.hpp>
+#include <opencv2/gapi/imgproc.hpp>
+#include <opencv2/gapi/infer.hpp>
+#include <opencv2/gapi/infer/parsers.hpp>
+#include <opencv2/gapi/render.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+#include <opencv2/highgui.hpp>
+
+#include <opencv2/gapi/oak/oak.hpp>
+#include <opencv2/gapi/oak/infer.hpp>
+
+const std::string keys =
+    "{ h help              |             | Print this help message }"
+    "{ detector            |             | Path to compiled .blob face detector model }"
+    "{ duration            | 1           | Number of frames to pull from camera and run inference on }";
+
+#ifdef HAVE_OAK
```

Comment:
Not sure if this `#ifdef` at a right place, considering you including `oak.hpp` explicitly above.

If code compiles with NO OAK but WITH those headers included, do we need it here at all?